MV3 host permission install prompt (#1887)
mozilla/extension-workshop: src/content/documentation/develop/manifest-v3-migration-guide.md

### Host permissions

Host permissions are no longer defined in the manifest.json keys `permissions` or `optional_permissions`. Instead, they are defined in the [`host_permissions`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/host_permissions) and [`optional_host_permissions`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/optional_host_permissions) keys.

> From Firefox 127, host permissions listed in `host_permissions` and `content_scripts` are displayed in the install prompt. However, if an extension update grants new host permissions, these are not shown to the user (see [Firefox bug 1893232](https://bugzil.la/1893232)). The user may grant or revoke host permissions on an ad hoc basis, including those granted at installation.

Reviewer: ```suggestion
Unlike Manifest V2, requested host permissions are not always granted. Users may revoke them at any time. Until Firefox 126, they were not automatically granted at install time.
From Firefox 127, host permissions listed in `host_permissions` and `content_scripts` are displayed in the install prompt. However, if an extension update grants new host permissions, these are not shown to the user (see [Firefox bug 1893232](https://bugzil.la/1893232)). The user may grant or revoke host permissions on an ad hoc basis, including those granted at installation.
```

Pretended a sentence to explain MV2's behavior for a more obvious comparison of the difference.

> From Firefox 127, `host_permissions` for Manifest V3 extensions is considered a requested permission and displayed to users during installation. However, if an extension update grants new host permissions, these are not shown to the user (see [Firefox bug 1893232](https://bugzil.la/1893232)). The user may grant or revoke host permissions on an ad hoc basis, including those granted at installation.

Author: ```suggestion
From Firefox 127, host permissions listed in `host_permissions` and `content_scripts` are displayed in the install prompt. However, if an extension update grants new host permissions, these are not shown to the user (see [Firefox bug 1893232](https://bugzil.la/1893232)). The user may grant or revoke host permissions on an ad hoc basis, including those granted at installation.
```

### Host permissions

Host permissions are no longer defined in the manifest.json keys `permissions` or `optional_permissions`. Instead, they are defined in the [`host_permissions`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/host_permissions) and [`optional_host_permissions`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/optional_host_permissions) keys.

> From Firefox 127, `host_permissions` for Manifest V3 extensions is considered a required permission and displayed to users during installation. However, if an extension update grants new host permissions, these are not shown to the user (see [Firefox bug 1893232](https://bugzil.la/1893232)). The user may grant or revoke host permissions on an ad hoc basis, including those granted at installation.

Reviewer 2: I would say "is considered a requested permission" instead, we use "required" for mv2 permissions which are not optional, and can't be revoked by the user on an ad hoc basis like `host_permissions` can.

Author: ```suggestion
From Firefox 127, `host_permissions` for Manifest V3 extensions is considered a requested permission and displayed to users during installation. However, if an extension update grants new host permissions, these are not shown to the user (see [Firefox bug 1893232](https://bugzil.la/1893232)). The user may grant or revoke host permissions on an ad hoc basis, including those granted at installation.
```

> In Chrome and `host_permissions` in Manifest V3 are treated as optional permissions and, therefore, must be requested.

Reviewer 2: This is not the case (yet?), and since this behaviors can (and will) change, I would avoid using definitive language about what other browsers do.

> In Chrome and `host_permissions` in Manifest V3 are treated as optional permissions and, therefore, must be requested.

Author: @zombie OK, I'll remove this and rely on the link to the MDN documentation as the source of more info should the reader want it.

> 
> In Chrome and `host_permissions` in Manifest V3 are treated as optional permissions and, therefore, must be requested.

Author: ```suggestion
```
